app/src/main/java/com/nutomic/syncthingandroid/activities/MainActivity.java

```diff
@@ -475,28 +477,36 @@ private void setOptimalDrawerWidth(View drawerContainer) {
     /**
      * Displays dialog asking user to accept/deny usage reporting.
      */
-    private void showUsageReportingDialog() {
+    private void showUsageReportingDialog(RestApi restApi) {
         final DialogInterface.OnClickListener listener = (dialog, which) -> {
-            Options options = getApi().getOptions();
-            switch (which) {
-                case DialogInterface.BUTTON_POSITIVE:
-                    getApi().getSystemInfo(systemInfo -> {
-                        options.urAccepted = systemInfo.urVersionMax;
-                    });
-                    break;
-                case DialogInterface.BUTTON_NEGATIVE:
-                    options.urAccepted = Options.USAGE_REPORTING_DENIED;
-                    break;
-                case DialogInterface.BUTTON_NEUTRAL:
-                    Uri uri = Uri.parse("https://data.syncthing.net");
-                    startActivity(new Intent(Intent.ACTION_VIEW, uri));
-                    break;
+            if (restApi == null) {
+                Toast.makeText(MainActivity.this, R.string.syncthing_terminated, Toast.LENGTH_SHORT).show();
+                return;
+            }
+            try {
+                switch (which) {
+                    case DialogInterface.BUTTON_POSITIVE:
+                        restApi.setUsageReporting(true);
+                        restApi.restart();
+                        break;
+                    case DialogInterface.BUTTON_NEGATIVE:
+                        restApi.setUsageReporting(false);
+                        restApi.restart();
+                        break;
+                    case DialogInterface.BUTTON_NEUTRAL:
+                        Uri uri = Uri.parse("https://data.syncthing.net");
+                        startActivity(new Intent(Intent.ACTION_VIEW, uri));
+                        break;
+                }
+            } catch (Exception e) {
```

Comment:
again, not a fan of the catch all situation, but I guess it's better than crashing.

Reply:
Yes, I agree with you on that.